diff --git a/ReadEtextsActivity.py b/ReadEtextsActivity.py
index 5615318..d590c36 100644
--- a/ReadEtextsActivity.py
+++ b/ReadEtextsActivity.py
@@ -632,7 +632,7 @@ class ReadEtextsActivity(SugarCompatibleActivity):
         textbuffer = self.textview.get_buffer()
         begin, end = textbuffer.get_selection_bounds()
         copy_text = textbuffer.get_text(begin, end, True)
-        self.clipboard.set_text(copy_text)
+        self.clipboard.set_text(copy_text, len(copy_text))
 
     def view_toolbar_go_fullscreen_cb(self, view_toolbar):
         self.fullscreen()
@@ -1146,10 +1146,11 @@ class ReadEtextsActivity(SugarCompatibleActivity):
         sel = selection.get_selected()
         if sel:
             model, _iter = sel
-            self.selected_title = model.get_value(_iter, COLUMN_TITLE)
-            self.selected_author = model.get_value(_iter, COLUMN_AUTHOR)
-            self.selected_path = model.get_value(_iter, COLUMN_PATH)
-            self.books_toolbar.enable_button(True)
+            if _iter:
+                self.selected_title = model.get_value(_iter, COLUMN_TITLE)
+                self.selected_author = model.get_value(_iter, COLUMN_AUTHOR)
+                self.selected_path = model.get_value(_iter, COLUMN_PATH)
+                self.books_toolbar.enable_button(True)
 
     def find_books(self, search_text):
         self.clear_downloaded_bytes()
