diff --git a/ReadEtextsActivity.py b/ReadEtextsActivity.py
index 42283c7..acfc38a 100644
--- a/ReadEtextsActivity.py
+++ b/ReadEtextsActivity.py
@@ -631,7 +631,7 @@ class ReadEtextsActivity(SugarCompatibleActivity):
         textbuffer = self.textview.get_buffer()
         begin, end = textbuffer.get_selection_bounds()
         copy_text = textbuffer.get_text(begin, end, True)
-        self.clipboard.set_text(copy_text)
+        self.clipboard.set_text(copy_text, len(copy_text))
 
     def view_toolbar_go_fullscreen_cb(self, view_toolbar):
         self.fullscreen()
@@ -918,7 +918,7 @@ class ReadEtextsActivity(SugarCompatibleActivity):
         linecount = 0
         label_text = '\n\n\n'
         while linecount < PAGE_SIZE:
-            line = self.etext_file.readline()
+            line = self.etext_file.readline().decode('iso-8859-1')
             if not line:
                 break
             else:
@@ -1327,7 +1327,7 @@ class ReadEtextsActivity(SugarCompatibleActivity):
         self.current_found_item = -1
         self.etext_file.seek(0)
         while self.etext_file:
-            line = str(self.etext_file.readline())
+            line = str(self.etext_file.readline().decode('iso-8859-1'))
             line_length = len(line)
             if not line:
                 break
